eng/bomgenerator/src/main/java/com/azure/tools/bomgenerator/models/BOMReport.java

```diff
@@ -0,0 +1,112 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package com.azure.tools.bomgenerator.models;
+
+import java.io.BufferedReader;
+import java.io.BufferedWriter;
+import java.io.File;
+import java.io.FileWriter;
+import java.io.IOException;
+import java.io.InputStreamReader;
+import java.time.LocalDateTime;
+import java.time.OffsetDateTime;
+import java.time.format.DateTimeFormatter;
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+public class BOMReport {
+    private final StringBuilder sb;
+    Map<BomDependency, List<ConflictingDependency>> dependencyConflicts;
+    private final String reportFileName;
+
+    public BOMReport(String reportFileName) {
+        sb = new StringBuilder();
+        dependencyConflicts = new HashMap<>();
+        this.reportFileName = reportFileName;
+    }
+
+    public void insertConflict(BomDependency dependencyWithConflict, ConflictingDependency conflictingDependency) {
```

Comment:
Can the `dependencyConflicts` map be passed in as a constructor arg instead of building the map here?